### PR TITLE
[OTX/XAI] MPA cls support baseline

### PR DESCRIPTION
**Pull request by dongkwan-kim01**
This is the same as #76 but reworked because of some commit history failure

---

**Summary**
Support explain mode to MPA classification

**Added**
- Add ClsExplainer class to mpa/cls/explainer.py
- Add explaination stage

**Changes**
- file name changes to `mpa/modules/hooks/auxiliary_hooks.py` -> `mpa/modules/hooks/recording_forward_hooks.py`
- Each explainer hook inherited from common base hook
- Renamed `SaliencyMapHook` -> `ActivationMapHook`
- `SaliencyMapHook` by default -> To use hook selector for explainer selection

**Test Codes**
MPA standalone code will be deprecated
Can test with OTX CLI(please refer OTX cli support, [link](https://github.com/openvinotoolkit/training_extensions/pull/1342))

### CODE REVIEW

**Comment by dongkwan-kim01** [comment]
@goodsong81 @harimkang @negvet Hello, because of some commit history error, I deleted the previous pr and re opened it. And previous comments applied, could you please re-check this PR?

**Comment by goodsong81** [comment]
Why did you do this? I cannot track the review history. Am I supposed to review there and comment here?

File: mpa/cls/explainer.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        - Run explain via hooks in recording_forward_hooks
+        """
+        self._init_logger()
+        explainer = kwargs.get('explainer')
+        self.explainer_hook = EXPLAINER_HOOK_SELECTOR.get(explainer.lower(), None)
+        if self.explainer_hook is None:
+            raise NotImplementedError(f'explainer algorithm {explainer} not supported')
+        logger.info(
+            f'explainer algorithm: {explainer}'
+        )
```
Comment (goodsong81) [inline mpa/cls/explainer.py:41]: Nit: Use capital letters for messages.

**Review by goodsong81 — APPROVED** [review]
LGTM.

**Pull request merged**